### ISSUE TITLE
official style guide for nushell script files and `nufmt` to auto-format such input

### ISSUE THREAD

**Issue by jokeyrhyme** [nushell/nushell#3938]
### Related problem

I'd like to write `.nu` script files and have white-space and other stylistic concerns handled for me in a consistent way

For a start, looking at the contributed scripts here ( https://github.com/nushell/nu_scripts ) I can see they are indented with varying numbers of spaces

### Describe the solution you'd like

- a first-party statement regarding recommended code style for nushell scripts, whether that be one line that says "as much as possible, do whatever upstream Rust does" or something more specific to nushell

- offer a first-party utility similar to [`gofumpt`](https://github.com/mvdan/gofumpt), `rustfmt`, or [`prettier`](https://prettier.io/) that handles not just indentation and spaces between tokens, but is also opinionated about line-length and thus also auto-line-breaks (just like these other 3x tools do for their own target formats)

### Describe alternatives you've considered

The status quo is honestly not urgent or even an inconvenience, I just like opinionated first-party code style guides :)

For now, I can just do whatever I like in my nushell code style no matter how different it is to other nushell scripts

### Additional context and details

Thanks so much for working on nushell and sharing it, it's been exciting and fun to follow along and try it out

**Comment by fdncred** [comment]
Agreed. `nufmt` would be awesome! We'd love to have this too. I looked into it recently to add to the vscode nushell plugin. Ugh, it looks like it either needs to be written in typescript, or we write something else and shell out and call that executable to format code.

**Comment by efx** [comment]
I would also appreciate this!
I was wondering today about the meaning of semicolons in various nu scripts.

A quality formatter would help teach me the semantics of nu as a language.

**Comment by jntrnr** [comment]
`nufmt` is now in-progress. With luck, it'll mature to the point of being a supported utility in the nushell toolbox.

**Comment by amtoine** [comment]
> `nufmt` is now in-progress. With luck, it'll mature to the point of being a supported utility in the nushell toolbox.
>
`nufmt` is :point_right: [here](https://github.com/nushell/nufmt) :point_left:
